### PR TITLE
Use more real world examples in BenchmarkSerializeObject

### PR DESCRIPTION
**Pull request by mborsz**
#### What this PR does / why we need it:

It changes BenchmarkSerializeObject to use slightly more real-world example:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 1000
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
        ports:
        - containerPort: 80
        resources:
          requests:
            cpu: "1m"
```

And then dumped one of pods as a json.

To be clear: This example is still far from perfect example (as in real world clusters we expect to see some configmaps, more labels, secrets, volumes etc), but at least has some pod-like structure (status, conditions, uuids).

Current result:
```
goos: linux
goarch: amd64
pkg: k8s.io/apiserver/pkg/endpoints/handlers/responsewriters
BenchmarkSerializeObject1000PodsPB
    writers_test.go:458: Payload size: 2629726, expected output size: 214478, ratio: 0.08
    writers_test.go:458: Payload size: 2629726, expected output size: 214452, ratio: 0.08
BenchmarkSerializeObject1000PodsPB-12        	      84	  15573340 ns/op	  640927 B/op	      44 allocs/op
BenchmarkSerializeObject10000PodsPB
    writers_test.go:458: Payload size: 26297133, expected output size: 2128756, ratio: 0.08
    writers_test.go:458: Payload size: 26297133, expected output size: 2128734, ratio: 0.08
    writers_test.go:458: Payload size: 26297133, expected output size: 2128705, ratio: 0.08
BenchmarkSerializeObject10000PodsPB-12       	       9	 132588147 ns/op	 5080175 B/op	      49 allocs/op
BenchmarkSerializeObject100000PodsPB
    writers_test.go:458: Payload size: 262971335, expected output size: 21263994, ratio: 0.08
BenchmarkSerializeObject100000PodsPB-12      	       1	1191164149 ns/op	80518208 B/op	      80 allocs/op
BenchmarkSerializeObject1000PodsJSON
    writers_test.go:458: Payload size: 4118115, expected output size: 256510, ratio: 0.06
    writers_test.go:458: Payload size: 4118115, expected output size: 256456, ratio: 0.06
BenchmarkSerializeObject1000PodsJSON-12      	      50	  22338815 ns/op	  647328 B/op	      43 allocs/op
BenchmarkSerializeObject10000PodsJSON
    writers_test.go:458: Payload size: 41180777, expected output size: 2548373, ratio: 0.06
    writers_test.go:458: Payload size: 41180777, expected output size: 2548031, ratio: 0.06
BenchmarkSerializeObject10000PodsJSON-12     	       6	 203910934 ns/op	10106026 B/op	      52 allocs/op
BenchmarkSerializeObject100000PodsJSON
    writers_test.go:458: Payload size: 411808348, expected output size: 25466926, ratio: 0.06
BenchmarkSerializeObject100000PodsJSON-12    	       1	1792364510 ns/op	80518192 B/op	      79 allocs/op
PASS
ok  	k8s.io/apiserver/pkg/endpoints/handlers/responsewriters	18.580s
```

/assign @wojtek-t 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mborsz** [comment]
@wojtek-t, as discussed offline: I changed this PR to avoid uploading 2.5MB serialized protobuf. Instead I upload a pod template and modify fields that differs between pods in the same deployment (generated using diff). It seems to work pretty well.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go [hunk 1]
```diff
@@ -391,18 +480,24 @@ func benchmarkSerializeObject(b *testing.B, size int) {
 	}
 }
 
-func BenchmarkSerializeObject10KB(b *testing.B) {
-	benchmarkSerializeObject(b, 10*1024)
+func BenchmarkSerializeObject1000PodsPB(b *testing.B) {
+	benchmarkSerializeObject(b, toProtoBuf(b, benchmarkItems(b, "testdata/pod.json", 1000)))
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go:484]: You may want to reset the time after computing benchmark items, something like:

items := benchmarkItems(...)
b.ResetTimer()
benchmarkSerializeObject(...)

**Review by wojtek-t — COMMENTED** [review]
This is great - just one comment.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go:484]: Ok - it's done later anyway. Never mind.

**Comment by wojtek-t** [comment]
This is great!

/kind cleanup

**Pull request merged**